pkg/sql/conn_executor.go

```diff
@@ -3285,6 +3285,23 @@ func errIsRetriable(err error) bool {
 		descs.IsTwoVersionInvariantViolationError(err)
 }
 
+// convertRetriableErrorIntoUserVisibleError converts internal retriable
```

Comment:
when and why would we want to convert an internal error into a user visible error? An example usage here would go a long way!

Reply:
Good point updated the comment:

// convertRetriableErrorIntoUserVisibleError converts internal retriable
// errors into external, so that the client goes and retries this
// transaction. One example of this is two version invariant errors, which
//happens when a schema change is waiting for a schema change transition to
// propagate. When this happens, we either need to retry externally or internally,
// depending on if we are in an explicit transaction.